elliptic_curve/src/point.rs

```diff
@@ -1,21 +1,49 @@
 use std::ops::Add;
-use finite_field::ff::FiniteField as FF;
+use finite_field::ff::PrimeField as FF;
+use finite_field::ff::FiniteField;
 use num_bigint::BigUint;
 use num_traits::{Zero, One};
 
+pub trait Point:
+    Sized
+    + PartialEq
+    + Add<Self, Output=Self>
+{
+    /// create a new point
+    fn new(x: Option<FF>, y: Option<FF>, a: FF, b: FF) -> Self;
+
+    /// return x
+    fn x(&self) -> FF;
+
+    /// return y
+    fn y(&self) -> FF;
+
+    /// return (x, y)
+    fn xy(&self) -> (FF, FF);
+}
+
 #[derive(Debug, Clone)]
-pub struct Point {
+pub struct ECCPoint {
     pub x: Option<FF>,
     pub y: Option<FF>,
     pub a: FF,
     pub b: FF,
 }
 
-impl Point {
-    pub fn new(x: Option<FF>, y: Option<FF>, a:FF, b:FF) -> Self {
+impl Point for ECCPoint {
+    fn new(x: Option<FF>, y: Option<FF>, a:FF, b:FF) -> Self {
         if x.is_none() && y.is_none() {
             return Self {x, y, a, b};
         } 
+
+        if x.is_none() && y.is_some() {
+            panic!("Invalid point");
+        }
+
+        if y.is_none() && x.is_some() {
+            panic!("Invalid point");
+        }
+
         if y.clone().unwrap().pow(2) != x.clone().unwrap().pow(3) 
```

Comment:
also consider returning a `Result` if the method is fallible instead of `unwrap`ing

https://stackoverflow.com/questions/39477684/should-i-avoid-unwrap-in-production-application